src/itype.h

```diff
@@ -589,6 +602,18 @@ struct itype {
     /** Actions an instance can perform (if any) indexed by action type */
     std::map<std::string, use_function> use_methods;
 
+    /** Default countdown interval (if any) for item */
+    int countdown_interval = 0;
+
+    /** Action to take when countdown expires */
+    use_function countdown_action;
+
+    /** Is item destroyed after the countdown action is run? */
+    bool countdown_destroy = false;
+
+    /** Fields to emit when item is in active state (id, qty, density) */
+    std::vector<std::tuple<field_id, int, int>> emit;
```

Comment:
Also, an explicit structure could have its own loading function and be used in more places: `explosion` actor, monster death etc.